sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1003,6 +1028,13 @@ static int mc_probe(struct platform_device *pdev)
 	card->dev = &pdev->dev;
 	snd_soc_card_set_drvdata(card, ctx);
 
+	card->components = devm_kasprintf(card->dev, GFP_KERNEL,
+					  "cfg-spk:%d cfg-amp:%d",
+					  (sof_sdw_quirk & SOF_SDW_FOUR_SPK)
+					  ? 4 : 2, amp_num);
```

Comment:
amp_num is zero here,  please check the original code to find how it is calculated